### PR TITLE
Move LoadRestRoutes into service-specific bootstrap handler

### PR DESCRIPTION
**Pull request by michaelestrin**
Fixes: https://github.com/edgexfoundry/edgex-go/issues/2263

Signed-off-by: Michael Estrin <m.estrin@dell.com>

### CODE REVIEW

**Comment by michaelestrin** [comment]
Dependent on https://github.com/edgexfoundry/go-mod-bootstrap/pull/25

**Comment by michaelestrin** [comment]
recheck

**Comment by michaelestrin** [comment]
Dependent on https://github.com/edgexfoundry/go-mod-bootstrap/pull/26

File: cmd/sys-mgmt-agent/main.go [hunk 1]
```diff
@@ -54,7 +56,9 @@ func main() {
 			return configuration
 		},
 	})
-	httpServer := httpserver.NewBootstrap(agent.LoadRestRoutes(dic))
+
+	router := mux.NewRouter()
```
[the remark this answers is not in the document]
Comment (michaelestrin) [reply cmd/sys-mgmt-agent/main.go]: Why would it?

File: internal/support/scheduler/init.go [hunk 2]
```diff
@@ -27,14 +27,25 @@ import (
...
+	}
+}
+
 // BootstrapHandler fulfills the BootstrapHandler contract and performs initialization needed by the scheduler service.
-func BootstrapHandler(
-	ctx context.Context,
-	wg *sync.WaitGroup,
-	startupTimer startup.Timer,
-	dic *di.Container) bool {
+func (b *Bootstrap) BootstrapHandler(ctx context.Context, wg *sync.WaitGroup, _ startup.Timer, dic *di.Container) bool {
```
[the remark this answers is not in the document]
Comment (michaelestrin) [reply internal/support/scheduler/init.go]: Not sure why this is explicitly called out...?

File: cmd/sys-mgmt-agent/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (akramtexas) [reply cmd/sys-mgmt-agent/main.go]: Never mind (I was playing the devil's advocate here...). Other core services (`metadata`, `command`, etc.) _do_ require it but the `SMA` does not 👍

**Comment by akramtexas** [comment]
> What blackbox test failures are you seeing?

- Seeing the following:

```
/usr/local/bin/newman run bin/postman-test/collections/support-scheduler.postman_collection.json -e bin/postman-test/environment/support-scheduler.postman_environment.json --folder intervalAction -d bin/postman-test/data/intervalActionData.json --disable-unicode

 1.  AssertionError  Status code is 200                                                             
                     expected false to be truthy                                                    
                     at assertion:0 in test-script                                                  
                     inside "intervalAction / 361 http://localhost:48085/api/v1/intervalaction/:id" 
                                                                                                    
 2.  AssertionError  Status code is 200                                                             
                     expected false to be truthy                                                    
                     at assertion:0 in test-script                                                  
                     inside "intervalAction / 362 http://localhost:48085/api/v1/intervalaction"     
                                                                                                    
 3.  AssertionError  Status code is 200                                                             
                     expected false to be truthy                                                    
                     at assertion:0 in test-script                                                  
                     inside "intervalAction / 363 http://localhost:48085/api/v1/intervalation"      
                                                                                                    
 4.  AssertionError  Status code is 200                                                             
                     expected false to be truthy                                                    
                     at assertion:0 in test-script                                                  
                     inside "intervalAction / 364 http://localhsot:48085/api/v1/intervalaction"  
```

```
/usr/local/bin/newman run bin/postman-test/collections/core-metadata.postman_collection.json -e bin/postman-test/environment/core-metadata.postman_environment.json --folder devicereport -d bin/postman-test/data/deviceReportData.json --disable-unicode

  #  failure         detail                                                                                         
                                                                                                                    
 1.  AssertionError  Status code is 200                                                                             
                     expected false to be truthy                                                                    
                     at assertion:0 in test-script                                                                  
                     inside "devicereport / 166 http://localhost:48081/api/v1/devicereport/id/{DeviceReportId}"     
                                                                                                                    
 2.  AssertionError  Status code is 200                                                                             
                     expected false to be truthy                                                                    
                     at assertion:0 in test-script                                                                  
                     inside "devicereport / 169 http://localhost:48081/api/v1/devicereport/name/{DeviceReportName}" 
```

```
/usr/local/bin/newman run bin/postman-test/collections/core-metadata.postman_collection.json -e bin/postman-test/environment/core-metadata.postman_environment.json --folder command -d bin/postman-test/data/commandData.json --disable-unicode


  #  failure         detail                                                                        
                                                                                                   
 1.  AssertionError  Status code is 200                                                            
                     expected false to be truthy                                                   
                     at assertion:0 in test-script                                                 
                     inside "command / 342 http://localhost:48081/api/v1/command/:id"              
                                                                                                   
 2.  AssertionError  Status code is 200                                                            
                     expected false to be truthy                                                   
                     at assertion:0 in test-script                                                 
                     inside "command / 345 http://localhost:48081/api/v1/command/device/:deviceId" 
```

**Comment by michaelestrin** [comment]
I'm not seeing these failures when I run the blackbox tests locally.

**Comment by akramtexas** [comment]
> I'm not seeing these failures when I run the blackbox tests locally.

The `PR` LGTM. Approving `PR`.

**Pull request merged**